flowey/flowey_lib_hvlite/src/init_cross_arch_build.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Install dependencies and set environment variables necessary for cross
+//! compiling to another architecture.
+
+use flowey::node::prelude::*;
+use std::collections::BTreeMap;
+
+flowey_request! {
+    pub struct Request{
+        pub target: target_lexicon::Triple,
+        pub cross_deps_write: WriteVar<SideEffect>,
+        pub injected_env_write: Option<WriteVar<BTreeMap<String, String>>>,
+    }
+}
+
+new_flow_node!(struct Node);
+
+impl FlowNode for Node {
+    type Request = Request;
+
+    fn imports(ctx: &mut ImportCtx<'_>) {
+        ctx.import::<flowey_lib_common::install_apt_pkg::Node>();
+    }
+
+    fn emit(requests: Vec<Self::Request>, ctx: &mut NodeCtx<'_>) -> anyhow::Result<()> {
+        for Request {
+            target,
+            cross_deps_write,
+            injected_env_write,
+        } in requests
+        {
+            let mut pre_build_deps = Vec::new();
+            let mut injected_env = BTreeMap::new();
+
```

Comment:
can you add a small optimization here? make this entire node a noop if the `target` and `ctx.arch()` match?